template/pkg/config.sls

```diff
@@ -1,11 +1,11 @@
 # -*- coding: utf-8 -*-
 # vim: ft=sls
 
-{%- from tpldir ~ "/map.jinja" import template with context %}
-{%- from tpldir ~ "/macros.jinja" import files_switch with context %}
+{%- from salt['file.dirname'](tpldir) ~ "/map.jinja" import template with context %}
```

Comment:
[Documentation shows usage of this format using a `+` instead of `~`](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.file.html#salt.modules.file.dirname). Even if both formats are correct, shouldn't we try to be consistent with the documented format?

Reply:
From the upstream-upstream documentation(!), i.e. the main Jinja docs.

http://jinja.pocoo.org/docs/2.10/templates/#math:

> `+`
> Adds two objects together. Usually the objects are numbers, but if both are strings or lists, you can concatenate them this way. **This, however, is not the preferred way to concatenate strings!** For string concatenation, have a look-see at the `~` operator. `{{ 1 + 1 }}` is 2.

http://jinja.pocoo.org/docs/2.10/templates/#other-operators:

> `~`
> Converts all operands into strings and concatenates them.
> 
> `{{ "Hello " ~ name ~ "!" }}` would return (assuming name is set to `'John'`) `Hello John!`.

* Emphasis above is mine.
* So in reality, the documentation you linked to needs to be updated.

Reply:
Yeah, I personally use `~` myself. Made the comment because read it in the docs and made sense to follow the doc. But you made more sense that the doc :smile: so let's go with `~`. Thanks!